src/data/bucket/line_bucket.js

```diff
@@ -319,16 +319,24 @@ class LineBucket implements Bucket {
         const cap = layout.get('line-cap').evaluate(feature, {});
         const miterLimit = layout.get('line-miter-limit');
         const roundLimit = layout.get('line-round-limit');
+        const dashArray = this.layers[0].paint.get('line-dasharray').value;
```

Comment:
Don't we need to check all layers to see if at least one of them has these properties? In gl-native I wrote a check for gradients which loops through the layers in the constructor of the bucket and sets a flag if any of them has it. 
Although I'm not fully sure if it's needed.

https://github.com/mapbox/mapbox-gl-native-internal/blob/internal/src/mbgl/renderer/buckets/line_bucket.cpp#L30

Reply:
Yes! I was confused by this line: `layout = this.layers[0].layout` Maybe all layers point to the same singleton layout?

https://github.com/mapbox/mapbox-gl-js/blob/f28733a812274dd25e838200b7bd7cf4bf20095f/src/data/bucket/line_bucket.js#L317

Exactly what needs to be checked was my main question in this PR. I'll take a look at your code and work through this again to see what needs to be checked here.

Reply:
@rreusser code has changed a bit, but afaik, layers are grouped by layout https://github.com/mapbox/mapbox-gl-js/blob/10961e05522d38bae952d7f01947fe6bfd6f2aa1/src/style/style_layer_index.js#L53.
@endanke is right, we might need to go through paint properties of all layers in a group.